helix-term/src/commands.rs

```diff
@@ -2116,31 +2116,33 @@ fn global_search(cx: &mut Context) {
                 let picker = FilePicker::new(
                     all_matches,
                     current_path,
-                    move |cx, FileResult { path, line_num }, action| {
-                        match cx.editor.open(path, action) {
-                            Ok(_) => {}
-                            Err(e) => {
-                                cx.editor.set_error(format!(
-                                    "Failed to open file '{}': {}",
-                                    path.display(),
-                                    e
-                                ));
+                    move |cx, file_result, action| {
+                        if let Some(FileResult { path, line_num }) = file_result {
```

Comment:
For picker callbacks with large bodies, you can avoid adding to the indentation and complicating the diff by using let..else to return early:

```rust
let Some(FileResult { path, line_num }) = file_result else { return; };
```

This isn't necessary for callback functions with small (1-3 line) bodies though, the `if let` is fine for those

Reply:
I missed that comment last time. It is fixed now